ansible/playbooks/roles/kafka_exporter/tasks/upgrade/main.yml

```diff
@@ -0,0 +1,96 @@
+---
+- name: Upgrade | Kafka Exporter | Check for upgrade flag file
+  stat:
+    path: "{{ lock_file }}"
+  register: lock_file_status
+
+- name: Upgrade | Kafka Exporter | Preflight check
+  when: not lock_file_status.stat.exists
+  block:
+  - name: Upgrade | Kafka Exporter | Create upgrade flag file
+    file:
+      path: "{{ lock_file }}"
+      state: touch
+
+  - name: Upgrade | Kafka Exporter | Get installed version
+    shell: >-
+      set -o pipefail &&
+      /opt/kafka_exporter/kafka_exporter --version &> >(grep -E kafka_exporter.*version | awk '{ print $3 }')
+    register: kafka_exporter_installed_version
+
+  - name: Upgrade | Kafka Exporter | Set installed version as variable
+    set_fact:
+      kafka_exporter_version_old: "{{ kafka_exporter_installed_version.stdout }}"
+
+  - name: Upgrade | Kafka Exporter | Avoid risk of downgrade
+    when: kafka_exporter_version_old is not version( kafka_exporter.version, '<' )
+    block:
+    - name: Upgrade | Kafka Exporter | Remove upgrade flag file
+      file:
+        path: "{{ lock_file }}"
+        state: absent
+
+    - name: Upgrade | Kafka Exporter | Fail task
```

Comment:
This will fail whole **upgrade** playbook - for me it is not necessary to fail and stop playbook execution (I mean, whole upgrade procedure) because we already have newer version.

Reply:
* I wanted to reference here part:
` 
  - name: Upgrade | Kafka Exporter | Fail task
     fail:
        msg: Kafka Exporter in newer version already installed!`

Reply:
done